components/common/CompleteButton.tsx

```diff
@@ -0,0 +1,34 @@
+'use client';
+
+import { Button } from '@/components/ui/button';
+import { PencilLine } from 'lucide-react';
+import { useState } from 'react';
+
+interface CompleteButtonProps {
```

Comment:
```suggestion
type Props = {
```

저 같은 경우는 props 타입은 위 처럼 type Props로 지정하는 편입니다! 우선 interface 와 type의 차이는 아래 와 같은데요.
![스크린샷 2025-04-04 오후 7 31 14](https://github.com/user-attachments/assets/42616c48-7b9b-4b0e-bd20-17faac0d79b2)

여러 차이점이 있지만 제가 type 을 주로 사용하는 가장 큰 이유는 선언병합 여부 입니다.
선언 병합(declaration merging)은 interface를 같은 이름으로 여러 번 선언하면 자동으로 합쳐지는 기능인데, 이게 때때로 문제가 될 수 있습니다.

```ts
// somewhere.ts
interface User {
  name: string;
}

// other.ts
interface User {
  age: number;
}

// 둘이 합쳐져서 → { name: string; age: number }
// 이 동작은 의도되지 않은 동작일 가능성이 높습니다.
```

Reply:
type으로 수정해두었습니다~! 앞으로도 이 방식으로 하겠습니다 :)